src/lib.rs

```diff
@@ -13,16 +12,27 @@ type CrustlsResult = c_int;
 pub const CRUSTLS_OK: c_int = 0;
 pub const CRUSTLS_ERROR: c_int = 1;
 
+// We use the opaque struct trick pattern to tell C about our types without
```

Comment:
nit: you refer to _the_ opaque struct trick as though I ought to know what is. If you have a reference to some documentation or a blog post explaining this technique, this comment would be a good place for it.

Reply:
Whoops, thanks for catching! I meant to include a link here. Will add.